src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -137,6 +138,13 @@ CHIP_ERROR DiscoveryImplPlatform::Advertise(const CommissionAdvertisingParameter
     char vendorSubType[8];
     const char * subTypes[3];
     size_t subTypeSize = 0;
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
```

Comment:
could we move this to a separate method?

this moves the method from 11 variables to 16 and that seems a bit long.

Reply:
i would still add 2 variables for the Rotating device id buffer & size, but i think it's a little cleaner.